src/cloudsc_gpu_atlas/CMakeLists.txt

```
@@ -0,0 +1,125 @@
+# (C) Copyright 1988- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+ecbuild_add_option( FEATURE CLOUDSC_GPU_SCC_ATLAS
+    DESCRIPTION "Build optimized GPU version of CLOUDSC based on Atlas + OpenACC" DEFAULT ON
+    CONDITION atlas_FOUND AND (Serialbox_FOUND OR HDF5_FOUND)
+)
+
+ecbuild_add_option( FEATURE CLOUDSC_GPU_SCC_HOIST_ATLAS
+    DESCRIPTION "Build optimized GPU version of CLOUDSC using SCC with hoisted temporary arrays (based on Atlas + OpenACC)" DEFAULT ON
+    CONDITION atlas_FOUND AND (Serialbox_FOUND OR HDF5_FOUND)
+)
+
+ecbuild_add_option( FEATURE CLOUDSC_GPU_SCC_K_CACHING_ATLAS
+    DESCRIPTION "Build optimized GPU version of CLOUDSC using SCC with k-caching of local array in the kernel (based on Atlas + OpenACC)" DEFAULT ON
+    CONDITION atlas_FOUND AND (Serialbox_FOUND OR HDF5_FOUND)
+)
+
+if( HAVE_CLOUDSC_GPU_SCC_ATLAS )
+    ecbuild_add_executable(
+        TARGET dwarf-cloudsc-gpu-scc-atlas-${prec}
+        SOURCES
+            cloudsc_global_atlas_state_mod.F90
+            expand_atlas_mod.F90
+            validate_atlas_mod.F90
+            ../cloudsc_gpu/cloudsc_gpu_scc_mod.F90
+            cloudsc_driver_scc_mod.F90
+            dwarf_cloudsc_atlas.F90
+        LIBS
+            atlas_f
+            pluto_f
+            cloudsc-common-${prec}-atlas
+        DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_SCC
+    )
+
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-gpu-scc-atlas
+        COMMAND bin/dwarf-cloudsc-gpu-scc-atlas
+        ARGS 1 1000 128
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 1
+        ENVIRONMENT "NVCOMPILER_ACC_CUDA_HEAPSIZE=64M"
+    )
+endif()
+
+if( HAVE_CLOUDSC_GPU_SCC_HOIST_ATLAS )
+    ecbuild_add_executable(
+        TARGET dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
+        SOURCES
+            cloudsc_global_atlas_state_mod.F90
+            expand_atlas_mod.F90
+            validate_atlas_mod.F90
+            ../cloudsc_gpu/cloudsc_gpu_scc_hoist_mod.F90
+            cloudsc_driver_scc_mod.F90
+            dwarf_cloudsc_atlas.F90
+        LIBS
+            atlas_f
+            pluto_f
+            cloudsc-common-${prec}-atlas
```

```suggestion
            cloudsc-common-atlas-${prec}
```
I'd assume that ${prec} should go at the end of the name. Of course also change everywhere used and defined.

---

This is the design of @reuterbal to have libraries, like atlas etc, attached to cloudsc-common after the precision. If @reuterbal agrees to change it in his build system, I have no problem changing it in the above CMake file.

---

Either is fine with me - I wasn't particularly strict about the naming scheme there, only trying to avoid any clashes and limiting the programming model exposure in the common libs.